### ISSUE TITLE
Adding Astronomical units

### ISSUE THREAD

**Issue by ebfortin** [angularsen/UnitsNet#679]
**Is your feature request related to a problem? Please describe.**
I'm working on a planet generation software and am using astronomical units all over. Right now these units are treated all as double. Which is exactly what UnistNet is there to resolve. 

**Describe the solution you'd like**
Astronomical units, like AU and mass ratio to solar mass, could be added to the relevant unit class like Length. 

**Describe alternatives you've considered**
Adding custom units. However this, as indicated, is not that well designed right now and I don't want to lose link with UnitsNet future releases. Also it makes sense to support these units used in a very large field of study.

**Comment by angularsen** [comment]
Hi and thanks for describing the feature. I think this sounds all very interesting and definitely something that belongs in the library. Would you be interested in doing one or more pull requests? I'm happy to assist.

https://github.com/angularsen/UnitsNet#want-to-contribute

**Comment by ebfortin** [comment]
For sure I would be interested. I will follow the link and prepare the code. I've checked how it is designed and adding what I want to the json files seem straightforward enough.

**Comment by angularsen** [comment]
Perfect! Look forward to it. Don't hesitate to ask if something is not clear.

**Comment by ebfortin** [comment]
Why is Power using decimal has its base type instead of double for almost all other units I've looked at?

**Comment by angularsen** [comment]
It's because it has a very wide range of units from very small to very large, so the conversions would fail due to poor precision because they have to convert via the base unit.
It's fixable, but for now there are around 3 quantities that use `decimal` internally as a workaround.

**Comment by angularsen** [comment]
Some discussions on improving precision:
- https://github.com/angularsen/UnitsNet/pull/588
- https://github.com/angularsen/UnitsNet/issues/478

**Comment by ebfortin** [comment]
Ok make sense. I now face a very Strange problem. I’ve Added Astronopmical Unit, Parsec and Light Years to Length.json. But when the test units are run, for Inches only, it fail because in the assert the au (Astronbomical Units) value is used. Any quick thought on what could cause that problem? Provenance : Courrier pour Windows 10 De : Andreas Gullberg LarsenEnvoyé le :17 juillet 2019 20:34À : angularsen/UnitsNetCc : Étienne Fortin; AuthorObjet :Re: [angularsen/UnitsNet] Adding Astronomical units (#679) It's because it has a very wide range of units from very small to very large, so the conversions would fail due to poor precision because they have to convert via the base unit.It's fixable, but for now there are around 3 quantities that use decimal internally as a workaround.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by ebfortin** [comment]
Also how can I change the precision for specific units. Provenance : Courrier pour Windows 10 De : Andreas Gullberg LarsenEnvoyé le :17 juillet 2019 20:36À : angularsen/UnitsNetCc : Étienne Fortin; AuthorObjet :Re: [angularsen/UnitsNet] Adding Astronomical units (#679) Some discussions on improving precision:#588#478—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by ebfortin** [comment]
Sorry I meant tolerance.  Provenance : Courrier pour Windows 10 De : efortin76@gmail.comEnvoyé le :17 juillet 2019 21:13À : angularsen/UnitsNet; angularsen/UnitsNetCc : AuthorObjet :RE: Re: [angularsen/UnitsNet] Adding Astronomical units (#679) Also how can I change the precision for specific units. Provenance : Courrier pour Windows 10 De : Andreas Gullberg LarsenEnvoyé le :17 juillet 2019 20:36À : angularsen/UnitsNetCc : Étienne Fortin; AuthorObjet :Re: [angularsen/UnitsNet] Adding Astronomical units (#679) Some discussions on improving precision:#588#478—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by angularsen** [comment]
Can you create a pull request so I can see what code changes you have? I don't see immediately what is wrong from your description.

Tolerance can be overridden by overriding the tolerance property values in the LengthTestBase class (or something like that), and setting it to something different than `1e-5` that I believe is the default.

**Comment by ebfortin** [comment]
I will create the pull Request. For the tolerance the problem is that the place where the property is defined is an auto generated class. So my change will be overriden the minute I execute generate-code.bat Provenance : Courrier pour Windows 10 De : Andreas Gullberg LarsenEnvoyé le :17 juillet 2019 21:20À : angularsen/UnitsNetCc : Étienne Fortin; AuthorObjet :Re: [angularsen/UnitsNet] Adding Astronomical units (#679) Can you create a pull request so I can see what code changes you have? I don't see immediately what is wrong from your description.Tolerance can be overridden by overriding the tolerance property values in the LengthTestBase class (or something like that), and setting it to something different than 1e-5 that I believe is the default.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by angularsen** [comment]
Sorry, I miswrote, you should edit `LengthTests` and override like this:

```c#
        protected override double YourUnitTolerance { get; }
```

This property is autogenerated in `LengthTestsBase`.

**Comment by angularsen** [comment]
Update: Forgot to specify the tolerance value.

**Comment by ebfortin** [comment]
Ok will do that. Pull Request created. Provenance : Courrier pour Windows 10 De : Andreas Gullberg LarsenEnvoyé le :17 juillet 2019 21:29À : angularsen/UnitsNetCc : Étienne Fortin; AuthorObjet :Re: [angularsen/UnitsNet] Adding Astronomical units (#679) Sorry, I miswrote, you should edit LengthTests and override like this:        protected override double YourUnitTolerance { get; }This property is autogenerated in LengthTestsBase.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by angularsen** [comment]
Fixed in #680
